src/android/run.ts

```diff
@@ -37,8 +42,11 @@ export async function run(args: string[]) {
   await waitForBoot(sdk, device);
 
   if (ports) {
-    await forwardPorts(sdk, device, ports);
-    log(`Forwarded device port ${ports.device} to host port ${ports.host}\n`);
+    ports.map(async (port: Ports) => {
```

Comment:
You should `await Promise.all(ports.map( ... ))` so the ports are forwarded before installing the apk.